### PR TITLE
Add eth_feeHistory RPC endpoint

### PR DESCRIPTION
**Pull request by mike6649**
### What was wrong?
Post-london RPC endpoint `eth_feeHistory` is not currently supported. This PR adds support for it with `fee_history` in the eth module.
Related to Issue #2038

### How was it fixed?
Following the [eth1.0-specs](https://github.com/ethereum/eth1.0-specs/blob/master/json-rpc/spec.json), `fee_history` function was added to the eth module.

### Todo:
- [x] Add support to async eth module
- [x] Testing
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

### CODE REVIEW

**Comment by mike6649** [comment]
Hi! Please let me know if there is any feedback to the implementation so far. I'm just roughly following the implementations of similar RPC methods, but I'm not sure whether the tests are written properly. Thanks all!

File: web3/_utils/module_testing/eth_module.py
Comment (fselmo) [inline web3/_utils/module_testing/eth_module.py:298]: @mike6649 just a small request to rename this to `test_eth_fee_history` to keep everything consistent

**Review by kclowes — COMMENTED** [review]
Thanks for the PR @mike6649! This looks like what we were going for! I added some nitpicky comments around the use of `apply_list_to_array_formatter` vs. `apply_formatter_to_array` that I'm curious to get your feedback on, but looks good other than that!

Also if you don't mind [adding a newsfragment](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md), that would be helpful. If you don't get to it, one of the maintainers can before merge. Thanks again!

**Comment by mike6649** [comment]
Hi @kclowes , I made the suggested changes above and added a newsfragment. Let me know if there is anything else! 👍

**Review by kclowes — APPROVED** [review]
Looks good to me! Thanks @mike6649! I'll get it merged soon!

**Pull request merged**
